packages/platform-server/test/incremental_hydration_spec.ts

```diff
@@ -923,189 +1023,346 @@ describe('platform-server partial hydration integration', () => {
-        globalThis.requestIdleCallback = nativeRequestIdleCallback;
-        globalThis.cancelIdleCallback = nativeCancelIdleCallback;
-        onIdleCallbackQueue.clear();
-        resetCounters();
+          expect(appHostNode.outerHTML).toContain('<span id="test">end</span>');
+        });
+
+        it('nested timer', async () => {
+          @Component({
+            selector: 'app',
+            template: `
+            <main (click)="fnA()">
+              @defer (on viewport; hydrate on interaction) {
+                <div id="main" (click)="fnA()">
+                  defer block rendered!
+                  @defer (on viewport; hydrate on timer(500)) {
+                    <article>
+                      <p id="nested">Nested defer block</p>
+                      <span id="test" (click)="fnB()">{{value()}}</span>
+                    </article>
+                  } @placeholder {
+                    <span>Inner block placeholder</span>
+                  }
+                </div>
+              } @placeholder {
+                <span>Outer block placeholder</span>
+              }
+            </main>
+          `,
+          })
+          class SimpleComponent {
+            value = signal('start');
+            fnA() {}
+            fnB() {
+              this.value.set('end');
+            }
+          }
+
+          const appId = 'custom-app-id';
+          const providers = [{provide: APP_ID, useValue: appId}];
+          const hydrationFeatures = () => [withIncrementalHydration()];
+
+          const html = await ssr(SimpleComponent, {envProviders: providers, hydrationFeatures});
+          const ssrContents = getAppContents(html);
+
+          // <main> uses "eager" `custom-app-id` namespace.
+          expect(ssrContents).toContain('<main jsaction="click:;');
+          // <div>s inside a defer block have `d0` as a namespace.
+          expect(ssrContents).toContain('<article>');
+          // Outer defer block is rendered.
+          expect(ssrContents).toContain('defer block rendered');
+
+          // Internal cleanup before we do server->client transition in this test.
+          resetTViewsFor(SimpleComponent);
+
+          ////////////////////////////////
+          const doc = getDocument();
+          const appRef = await prepareEnvironmentAndHydrate(doc, html, SimpleComponent, {
+            envProviders: [
+              ...providers,
+              {provide: PLATFORM_ID, useValue: 'browser'},
+              withDebugConsole(),
+            ],
+            hydrationFeatures,
+          });
+          const compRef = getComponentRef<SimpleComponent>(appRef);
+          appRef.tick();
+          await appRef.whenStable();
+
+          const appHostNode = compRef.location.nativeElement;
+
+          expect(appHostNode.outerHTML).toContain('<article>');
+
+          verifyHasLog(
+            appRef,
+            'Angular hydrated 1 component(s) and 21 node(s). 0 component(s) and 2 defer block(s) were skipped',
```

Comment:
The "2 defer block(s) were skipped" may sound like something was misconfigured. I'm wondering if we can phrase it like this instead (and avoid including an extra sentence if the number of such blocks is 0), WDYT?
```suggestion
            'Angular hydrated 1 component(s) and 21 node(s). 0 component(s) were skipped. 2 defer block(s) were configured to use incremental hydration.',
```